### PR TITLE
Provide tests for svginfo

### PR DESCRIPTION
**Pull request by JeremiePat**
Here are some test for the current state of {{svginfo}}

They cover:
- [x] calling svginfo in preview mode
- [x] calling svginfo for non existent SVG element
- [x] calling svginfo with an explicit parameter
- [x] calling svginfo with an implicit parameter (based on the page slug)
- [x] checking the various type of linking for categories and permitted contents

> **Note:** _This should be reviewed and merged before PR #670 that introduce changes to svginfo_

### CODE REVIEW

File: tests/macros/test-svginfo.js [hunk 1]
```diff
@@ -0,0 +1,496 @@
...
+// ----------------------------------------------------------------------------
+// Each test case is define by:
+// A `title` to make the test understandable by a human behing
+// An `input` which is an Array of parameters that will be passed to the macro
+// An `output` which is the string that the macro should return,
+// A `env` that is overriding the default env variable inside the macro
+//
+// NOTE: we could probably make that more generic by having a single test
+//       runner (see below) and a bunch of JSON files (one per macro) to
+//       describe all the possible inputs and their expected outputs.
```
Comment (escattone) [inline tests/macros/test-svginfo.js]: On further reflection, in order for this approach to be used for more general testing, the definition of each test case would also have to include its `beforeEachMacro` set-up (creating and populating the stubs for example), which seems difficult to standardize.

File: tests/macros/test-svginfo.js [hunk 2]
```diff
@@ -0,0 +1,496 @@
...
+        Object.keys(MOCK_PAGES['en-US']).forEach((key) => {
+            const { url, data } = MOCK_PAGES['en-US'][key];
+            macro.ctx.wiki.getPage.withArgs(url).returns(data);
+        });
+
+        Object.keys(MOCK_PAGES['zh-CN']).forEach((key) => {
+            const { url, data } = MOCK_PAGES['zh-CN'][key];
+            macro.ctx.wiki.getPage.withArgs(url).returns(data);
+        });
+    });
```
Comment (escattone) [inline tests/macros/test-svginfo.js]: This is nicely done, but it's missing the stub for the `wiki.pageExists` call that is made within the `SVGElement` macro (via the call to `template("SVGElement"...` made in one of the paths through the `svginfo` macro). The way the test runs currently, it's making an actual network request via the real `wiki.pageExists`. Something like this worked for me during my testing:
```javascript
    beforeEachMacro((macro) => {
        // let's make sure we have clean calls to wiki.getPage
        macro.ctx.wiki.getPage = sinon.stub();
        macro.ctx.wiki.pageExists = sinon.stub();

        Object.keys(MOCK_PAGES['en-US']).forEach((key) => {
            const { url, data } = MOCK_PAGES['en-US'][key];
            macro.ctx.wiki.getPage.withArgs(url).returns(data);
            macro.ctx.wiki.pageExists.withArgs(url).returns(true);
        });

        Object.keys(MOCK_PAGES['zh-CN']).forEach((key) => {
            const { url, data } = MOCK_PAGES['zh-CN'][key];
            macro.ctx.wiki.getPage.withArgs(url).returns(data);
            macro.ctx.wiki.pageExists.withArgs(url).returns(true);
        });
    });
```

This reminds me that I need to submit a PR that stubs the `request` package within the `macro.ctx` object, so that it throws an error when called. I think that would be the best way to catch all of the attempts to make real network calls, and help us discover what we need to stub.

File: tests/macros/test-svginfo.js [hunk 3]
```diff
@@ -0,0 +1,496 @@
...
+
+    // Set the list of permitted content that apply to the element
+    const permittedContent = [DESCRIPTION];
+
+    if (data.content.elements) {
+        // Regroup permitted content between named groups and standalone elements
+        let { elements, groups } = data.content.elements.reduce((acc, value) => {
+            if (value.indexOf('&lt;') !== -1) {
+                let key     = value.replace(/&lt;|&gt;/g, '');
+                let url     = URL(locale, SVG_BASE_SLUG, 'Element', key);
```
Comment (escattone) [inline tests/macros/test-svginfo.js]: Whitespace again: I would prefer only a single space before each `=`.

File: tests/macros/test-svginfo.js [hunk 4]
```diff
@@ -0,0 +1,496 @@
...
+                let summary = getSummary(key, locale);
+
+                acc.elements.push(`<a href="${url}"  title="${summary}"><code>${value}</code></a>`);
+            } else {
+                let anchor = "#" + camelToSnake(value);
+                let url    = URL(locale, SVG_BASE_SLUG, 'Element') + anchor;
+                let label  = _(value, locale);
```
Comment (escattone) [inline tests/macros/test-svginfo.js]: Whitespace again: I would prefer only a single space before each `=`.

File: tests/macros/test-svginfo.js [hunk 5]
```diff
@@ -0,0 +1,496 @@
...
+
+                acc.groups.push(`<a href="${url}">${label}</a>`);
+            }
+
+            return acc;
+        }, {elements: [], groups: []});
+
+        // Named groups must be listed first (each on new lines)
+        if (groups.length   > 0) permittedContent.push(groups.join('<br/>'));
```
Comment (escattone) [inline tests/macros/test-svginfo.js]: Whitespace again: I would prefer only a single space before the `>`.

File: tests/macros/test-svginfo.js (hunk 2 above)
Comment (escattone) [reply tests/macros/test-svginfo.js]: It's been a while since I've worked with the test framework, as I forgot that I had already stubbed the `request` package long ago in https://github.com/mdn/kumascript/pull/240! 🤦‍♂️

So your tests were not making real network calls, sorry! I still think it's best to create and populate the stub for `wiki.pageExists` instead of relying on whatever `wiki.pageExists` returns using the un-populated `request` stub.

File: tests/macros/test-svginfo.js (hunk 1 above)
Comment (JeremiePat) [reply tests/macros/test-svginfo.js]: Indeed, at that point I need to get into more testing to see what can be standardize (and how) and what cannot and how is it possible to neatly handle special edge case. Those notes are here to remember to have further discussion when we will have more tests.

**Comment by JeremiePat** [comment]
It should be better now.

For `wiki.pageExists` I follow @escattone recommendation as we do not need anything more clever than that in that case.

For the white space, I made the changes even if I'm someone who is very sensitive to vertical alignment but I don't want to fight on this.
